sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -400,6 +410,22 @@ static void appendCompileOptionsFromImage(std::string &CompileOpts,
     // metadata.
     CompileOpts += isEsimdImage ? "-doubleGRF" : "-ze-opt-large-register-file";
   }
+  // Add optimization flags
```

Comment:
A possible improvement here is to add a plugin API that can return a backend specific option for a given front-end option.

Reply:
I added a separate commit for this. Thanks